docs/components/file-upload.md

```diff
@@ -0,0 +1,213 @@
+    </KFileUpload>
+  </template>
+</KCard>
+
+<div class="mt-6">Emitted value: <pre v-if="imageName && imageSize" class="emitted-value">{{ `File Name: ${imageName}` }} and {{ `File Size: ${imageSize}` }}</pre></div>
+
+```html
+<KFileUpload appearance="image" icon="image" :accept="['.yml', '.yaml', '.json', '.md', '.markdown', 'image/gif', 'image/jpg', 'image/jpeg', 'image/ico', 'image/png']" placeholder="Upload new image (Max 4 MB)" />
+```
+
+### label
+
+String to be used as the `KFileUpload` label when `appearance` is of type `file`.
+
+### labelAttributes
+
+Use the `labelAttributes` prop to configure the **KLabel's** [props](/components/label.html) if using the `label` prop when `appearance` is of type `file`.
+
+<KCard>
+  <template v-slot:body>
+    <KFileUpload label="Upload File" :label-attributes="{ help: 'Hello, here is the example' }" icon="close" :accept="['.yml', '.yaml', '.json', '.md', '.markdown', 'image/gif', 'image/jpg', 'image/jpeg', 'image/ico', 'image/png']" appearance="file" removable />
+  </template>
+</KCard>
+
+```html
+<KFileUpload label="Upload File" :label-attributes="{ help: 'Hello, here is the example' }" icon="close" :accept="['.yml', '.yaml', '.json', '.md', '.markdown', 'image/gif', 'image/jpg', 'image/jpeg', 'image/ico', 'image/png']" appearance="file" removable />
+```
+
+### help
+
+Use the `help` prop to display text under KInput.
+
+<KCard>
+  <template v-slot:body>
+    <KFileUpload label="Upload File" :label-attributes="{ help: 'Hello, here is the example' }" help="Some text here.." icon="close" :accept="['.yml', '.yaml', '.json', '.md', '.markdown', 'image/gif', 'image/jpg', 'image/jpeg', 'image/ico', 'image/png']" appearance="file" removable />
+  </template>
+</KCard>
+
+```html
+<KFileUpload label="Upload File" help="Some text here.." icon="close" :accept="['.yml', '.yaml', '.json', '.md', '.markdown', 'image/gif', 'image/jpg', 'image/jpeg', 'image/ico', 'image/png']" appearance="file" removable />
+```
+
+
+### placeholder
+
+Use the `placeholder` prop to display placeholder text.
+
+<KCard class="mt-6">
+  <template v-slot:body>
+    <KFileUpload appearance="image" :accept="['.yml', '.yaml', '.json', '.md', '.markdown', 'image/gif', 'image/jpg', 'image/jpeg', 'image/ico', 'image/png']" placeholder="You can change the text here!" icon="kong" >
+    </KFileUpload>
+  </template>
+</KCard>
+
+```html
+<KFileUpload appearance="image" :accept="['.yml', '.yaml', '.json', '.md', '.markdown', 'image/gif', 'image/jpg', 'image/jpeg', 'image/ico', 'image/png']" placeholder="You can change the text here!" icon="kong" />
+```
+
+### removable
+
+A `cancel` button can be displayed for `file` type upload, by default this is set to `false`. This button is only visible once a file has been added.
```

Comment:
Again, I wouldn't restrict this prop to being used with `file` appearance. Users can choose whether or not they want to use it. Can we get an example?

Reply:
Also think it may be good to have this `true` by default as I think most cases are going to want the option to remove the file once it's been selected. What do you think?